In [1]:
import plotly.express as px
import pandas as pd

In [2]:
df = pd.read_csv("queries_results/avg_time_city_area.csv")

In [4]:
fig = px.bar(df, x='City', y='avg_time', title="Average Delivery Time by City Area in minutes", color='avg_time')
fig.show()

In [5]:
df = pd.read_csv("queries_results/avg_time_traffic.csv")
fig = px.bar(df, x='Road_traffic_density', y='avg_time', title="Traffic Impact on Delivery Time", color='avg_time')
fig.show()

In [7]:
df = pd.read_csv("queries_results/avg_time_vehicle_used.csv")
fig = px.bar(df, x='Type_of_vehicle', y='avg_time', title="Average Delivery Time by Vehicle Type", color='avg_time')
fig.show()

In [10]:
df = pd.read_csv('queries_results/time_intervals.csv')
fig = px.bar(df, x='time_interval', y='avg_time_taken', title="When is best to place your order", color='avg_time_taken')
fig.show()

In [12]:
df = pd.read_csv("queries_results/avg_time_traffic.csv")

In [13]:
print(df['Road_traffic_density'].unique())

['Jam ' 'High ' 'Medium ' 'Low ']


In [15]:
jam = df.loc[df['Road_traffic_density'] == 'Jam ', 'avg_time'].values[0]
low = df.loc[df['Road_traffic_density'] == 'Low ', 'avg_time'].values[0]

percent_increase = ((jam - low) / low) * 100
print(f"Percent increase: {percent_increase:.2f}%")

Percent increase: 45.25%


### Insights
- The type of vehicle used does not significantly impact delivery time – the maximum difference between the fastest and slowest vehicle is around 3 minutes, which is hardly noticeable to the customer

- Your location plays a major role in delivery time – orders delivered to semi-urban areas take nearly twice as long as those in urban areas

- Heavy traffic conditions significantly increase wait times – when traffic is jammed, delivery times are approximately 45% longer compared to low-traffic conditions

- Morning (6–11 AM) and night (12–5 AM) are the best times to place an order – these time intervals have the shortest average delivery durations, while evening is the slowest period, followed by the afternoon